libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1072,6 +1086,15 @@ AP_GPS_UBLOX::_parse_gps(void)
         next_fix = state.status;
 #endif
         break;
+    case MSG_TIMEGPS:
+        Debug("MSG_TIMEGPS");
+        _check_new_itow(_buffer.timegps.itow);
+        if (havePvtMsg) {
+            _unconfigured_messages |= CONFIG_RATE_SOL;
```

Comment:
You need to either add a flag for have time GPS, or unconfigure the SOL message if havePVT is true, since currently if you have PVT the TIMEGPS message will be there (at least going back to the 6 series). All this currently does is continually mark us as unconfigured which blocks arming at 1Hz until we've confirmed the rate again, which is done in a slow loop.

Reply:
ive changed it to set the sol message rateto  0 if havePvtMsg is set